spec/models/client_spec.rb

```diff
@@ -49,7 +49,9 @@ def app
     client = Client.create(name: 'Delete Me')
     expect(client).to exist
     expect { client.delete }.to_not raise_error
+    ### Test passes but the item's name and address can still be printed?
     expect(client).to_not exist
+    print client, ' ', client.name
```

Comment:
I'm 80% sure so. I think the behind the scenes logic is like this:

1. var client is stored in memory to a Client object with name = 'Delete Me'
2. client.delete is ran which is just a SQL execution on the DB
3. the print line is referring data from the variable in the memory and not the DB record.

I think the problem is the `().to exists` and `().to_not exist` which is doing something else than we are expecting.